File: 1299 - Replace Elements with Greatest Element on Right Side.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int arr_length = arr.size() - 1;
        int curr_max = -1;
        int placeholder = arr[arr_length];
        arr[arr_length] = curr_max;
        for(int i = arr_length - 1; i >= 0; i--){
            curr_max = max(curr_max, placeholder);
            placeholder = arr[i];
            arr[i] = curr_max;
        }
        return arr;
    }
};

File: 1299 - Replace Elements with Greatest Element on Right Side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        curr_max = -1
        placeholder = arr[-1]
        arr[-1] = -1
        for i in range(len(arr) - 2, -1, -1):
            curr_max = max(curr_max, placeholder)
            placeholder = arr[i]
            arr[i] = curr_max
        return arr